concepts/observability-configurable-logging/07-multiple-logpipeline-investigation.md

```diff
@@ -0,0 +1,136 @@
+
+Result:
+- When one of the outputs is down, the filesystem buffer is filled up and only keeps 150M of latest data (old data is discarded first). Eventually, when the filesystem buffer at tail plugin is full, the output plugin for Loki stops as well.
+- When both outputs are down, the tail plugin is stopped after filling the filesystem buffer, even though the buffer was not completely full (104M/150M).
+
+### Case 2
+![a](./assets/logpipeline-invstigation/case-2/case-2.svg)
+
+Setup:
+1. One input with two outputs (Loki (grafana-loki plugin) and mock server (http plugin)) with rewrite tags
+2. Two log pipelines:
+   - [Loki](./assets/logpipeline-invstigation/case-2/loki.yaml)
+   - [Mock server](./assets/logpipeline-invstigation/case-2/mockserver.yml)
+
+
+Result:
+- Output chunking keeps with 150M of newest data.
+- The tail plugin sends the data to the next phase of the pipeline (filter plugin). If the output is not working, only the filesystem buffer at the rewrite tag is filled and the tail plugin buffer isn't filled.
+- When the filesystem buffer is full, error logs state this fact, and that the old logs are discarded.
+    ```unix
+    [2022/04/21 14:38:23] [error] [input:emitter:log_emitter] error registering chunk with tag: log_rewritten
+    [2022/04/21 14:38:24] [error] [input chunk] chunk 1-1650551903.999375896.flb would exceed total limit size in plugin http.1
+    ```
+- When the Grafana-Loki plugin is stopped, it stops all the pipelines including mock server, even though that was working fine.
+- When mock server was down, the logs were still shipped to Loki.
+
+
+### Case 3
+
+Setup:
+1. One input with two outputs (mock server and mock server) with rewrite tags
+2. Two log pipelines:
+   - [mockserver-1](./assets/logpipeline-invstigation/case-3/mockserver-1.yml)
+   - [mockserver-2](./assets/logpipeline-invstigation/case-3/mockserver-2.yml)
+
+![a](./assets/logpipeline-invstigation/case-3/case-3.svg)
+Result:
+- When both outputs are down, the filesystem buffer was filled and eventually the Fluent Bit Pod went down with error code `500` (most probably because of CPU throttling).
+- Tail plugin kept pushing logs to the rewrite filesystem buffer, and they were eventually lost.
+### Case 4
+Setup:
+1. One input with two outputs (Loki (Loki plugin), and mock server (http plugin)) with rewrite tags
+2. Two log pipelines:
+   - [Loki](./assets/logpipeline-invstigation/case-4/loki.yml)
+   - [Mock server](./assets/logpipeline-invstigation/case-4/mock-server.yml)
+
+![a](./assets/logpipeline-invstigation/case-4/case-4.svg)
+
+Result:
+- Mock server was down. The Loki output kept working.
+- The chunks in the `filesystem buffer` are rolled: Old chunks are discarded; new ones created.
+- When Loki output was down, the mock server output kept working.
+
+![mockserver-down](/assets/logpipeline-invstigation/case-4/dashboard-mock-down.png)
+
+
+![loki-down](/assets/logpipeline-invstigation/case-4/dashboard-loki-down.png)
+
+### Case 5
```

Comment:
What is the difference to case 1? The other Loki Plugin? Since Case 2-5 are very similar (one input with rewrite_tag filter), maybe we could make it Case 2a, Case 2b and Case 2c. That would give a structure that is a bit easier to follow

Reply:
Maybe we could also keep the naming of the previous document:
- Setup 1 not tested here
- Setup 2 = Case 6
- Setup 3a-c = Case 2-6
- Setup 4 =  Case 1

Reply:
renamed.